buildSrc/src/main/java/io/opentelemetry/android/PublishPlugin.java

```diff
@@ -0,0 +1,112 @@
+package io.opentelemetry.android;
+
+import com.android.build.api.dsl.LibraryExtension;
+
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.plugins.PluginContainer;
+import org.gradle.api.publish.PublishingExtension;
+import org.gradle.api.publish.maven.MavenPublication;
+import org.gradle.plugins.signing.SigningExtension;
+
+import java.util.Locale;
+
+import kotlin.Unit;
+
+/**
+ * This plugin takes care of configuring a library project by creating a maven publication that
+ * contains the library artifact and (when the "-Prelease=true" param is present in the gradle command) it will also include
+ * the library's javadoc, sources artifacts and all the artifacts will be signed.
+ */
+public class PublishPlugin implements Plugin<Project> {
+    private Project project;
+
+    @Override
+    public void apply(Project project) {
+        this.project = project;
+        PluginContainer plugins = project.getPlugins();
+        applyPublishingPlugins(plugins);
+        plugins.withId("com.android.library", plugin -> {
+            configureAndroidLibraryPublication();
+        });
+    }
+
+    private void applyPublishingPlugins(PluginContainer plugins) {
+        plugins.apply("maven-publish");
+        plugins.apply("signing");
+    }
+
+    private void configureAndroidLibraryPublication() {
+        LibraryExtension androidExtension = project.getExtensions().getByType(LibraryExtension.class);
+        String variantToRelease = "release";
+        androidExtension.getPublishing().singleVariant(variantToRelease, librarySingleVariant -> {
+            if (isARelease()) {
+                librarySingleVariant.withJavadocJar();
+                librarySingleVariant.withSourcesJar();
+            }
+            return Unit.INSTANCE;
+        });
+
+        project.afterEvaluate(self -> addMavenPublication(variantToRelease));
+    }
+
+    private void addMavenPublication(String componentName) {
+        PublishingExtension mavenPublishExtension = project.getExtensions().getByType(PublishingExtension.class);
+        mavenPublishExtension.getPublications().create("maven", MavenPublication.class, publication -> {
+            publication.from(project.getComponents().findByName(componentName));
+            configurePom(publication);
+            if (isARelease()) {
+                signPublication(publication);
+            }
+        });
+    }
+
+    private void configurePom(MavenPublication publication) {
+        publication.pom(mavenPom -> {
+            String repoUrl = "https://github.com/open-telemetry/opentelemetry-android";
+            mavenPom.getName().set(project.getGroup() + ":" + project.getName());
```

Comment:
In the [instrumentation repo](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/conventions/src/main/kotlin/otel.publish-conventions.gradle.kts#L42) the name is the human-readable name, which I'd prefer to just be something like "OpenTelemetry for Android" or just "OpenTelemetry Android".

Reply:
Good call! I went with "OpenTelemetry Android".